js/lex.go

```diff
@@ -33,6 +33,30 @@ const (
 	TemplateToken
 )
 
+// TokenState determines a state in which next token should be read
+type TokenState uint32
+
+// TokenState values
+const (
+	ExprState TokenState = iota
+	StmtParensState
+	SubscriptState
+	PropNameState
+)
```

Comment:
So what do you mean with `subscript`? Anything that is a number, regexp or a string?